### PR TITLE
[MI-2483]:Fixed the URL error in config modal

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**

### CODE REVIEW

File: server/confluence_config.go [hunk 1]
```diff
@@ -28,14 +30,32 @@ func (p *Plugin) handleConfluenceConfig(w http.ResponseWriter, r *http.Request)
...
 		p.API.LogError("Error decoding the submit dialog request.", "Error", err.Error())
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
 
+	configInstanceURL := submitRequest.Submission[configServerURL].(string)
+
+	if vErr := utils.IsValidURL(configInstanceURL); vErr != nil {
+		response.Error = ErrorInvalidURL
+
```
Comment (ayusht2810) [inline server/confluence_config.go]:
```suggestion
```

File: server/utils/utils.go [hunk 2]
```diff
@@ -256,3 +256,21 @@ func Map(vs []string, f func(string) string) []string {
 	}
 	return vsm
 }
+
+// IsValidURL checks if a given URL is a valid URL with a host and a http or https scheme.
```
Comment (ayusht2810) [inline server/utils/utils.go:260]:
```suggestion
// IsValidURL checks if a given URL is a valid URL with a host and an http or https scheme.
```

File: server/utils/utils.go [hunk 3]
```diff
@@ -256,3 +256,21 @@ func Map(vs []string, f func(string) string) []string {
...
+func IsValidURL(rawURL string) error {
+	u, err := url.ParseRequestURI(rawURL)
+	if err != nil {
+		return err
+	}
+
+	if u.Scheme != "http" && u.Scheme != "https" {
+		return errors.Errorf("URL schema must either be %q or %q", "http", "https")
```
Comment (ayusht2810) [inline server/utils/utils.go]:
```suggestion
		return errors.Errorf("URL schema must either be http or https")
```

Comment (ayusht2810) [reply server/utils/utils.go]: can't we do this directly?

File: server/confluence_config.go [hunk 4]
```diff
@@ -28,14 +30,32 @@ func (p *Plugin) handleConfluenceConfig(w http.ResponseWriter, r *http.Request)
...
 	submitRequest := &model.SubmitDialogRequest{}
+	response := &model.SubmitDialogResponse{}
 	if err := decoder.Decode(&submitRequest); err != nil {
 		p.API.LogError("Error decoding the submit dialog request.", "Error", err.Error())
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
 
+	configInstanceURL := submitRequest.Submission[configServerURL].(string)
+
```
Comment (ayusht2810) [inline server/confluence_config.go]:
```suggestion
```

File: server/confluence_config.go [hunk 5]
```diff
@@ -28,14 +30,32 @@ func (p *Plugin) handleConfluenceConfig(w http.ResponseWriter, r *http.Request)
...
+	if vErr := utils.IsValidURL(configInstanceURL); vErr != nil {
+		response.Error = ErrorInvalidURL
+
+		response, mErr := json.Marshal(response)
+		if mErr != nil {
+			p.API.LogError("Error in marshalling the URL.", "Error", mErr.Error())
```
Comment (ayusht2810) [inline server/confluence_config.go]:
```suggestion
			p.API.LogError("Error in marshalling the response.", "Error", mErr.Error())
```

File: server/confluence_config.go [hunk 6]
```diff
@@ -28,14 +30,32 @@ func (p *Plugin) handleConfluenceConfig(w http.ResponseWriter, r *http.Request)
...
+			http.Error(w, mErr.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		p.API.LogError("Error in validating the URL.", "Error", vErr.Error())
```
Comment (ayusht2810) [inline server/confluence_config.go:50]: shouldn't the message be different? This error will occur when the link is not valid, but this statement means that we were not able to validate the link itself.

File: server/utils/utils.go (hunk 2 above)
Comment (Kshitij-Katiyar) [reply server/utils/utils.go:260]: i dont think we need this change

File: server/confluence_config.go (hunk 6 above)
Comment (Kshitij-Katiyar) [reply server/confluence_config.go:50]: This error could also come if we are unable to validate the URL due to any number of reasons that's why it's better to log this statement and the main error will be logged after that.I am open to suggestions if you have a better idea

File: server/utils/utils.go [hunk 7]
```diff
@@ -265,7 +265,7 @@ func IsValidURL(rawURL string) error {
 	}
 
 	if u.Scheme != "http" && u.Scheme != "https" {
-		return errors.Errorf("URL schema must either be %q or %q", "http", "https")
+		errors.New("URL schema must either be http or https")
```
Comment (ayusht2810) [inline server/utils/utils.go]: why removed the return from here?

File: server/confluence_config.go (hunk 6 above)
Comment (ayusht2810) [reply server/confluence_config.go:50]: this seems fine

**Review by ayusht2810 — APPROVED** [review]

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**